src/encoded/helpers/upload_credentials.py

```diff
@@ -1,8 +1,88 @@
 import boto3
 import botocore
+import copy
 import json
 
 
+EXTERNAL_BUCKET_STATEMENTS = [
+    {
+        'Action': 's3:GetObject',
+        'Resource': lambda s: 'arn:aws:s3:::%s/*' % s,
+        'Effect': 'Allow',
+    },
+    {
+        'Action': 's3:ListBucket',
+        'Resource': lambda s: 'arn:aws:s3:::%s' % s,
+        'Effect': 'Allow',
+    },
+    {
+        'Action': 's3:GetObjectAcl',
+        'Resource': lambda s: 'arn:aws:s3:::%s/*' % s,
+        'Effect': 'Allow',
+    }
+]
+
+
+def _compile_statements_from_list(buckets_list):
+    statements = []
+    if buckets_list:
+        for ext_policy in EXTERNAL_BUCKET_STATEMENTS:
+            new_policy = copy.copy(ext_policy)
+            new_policy['Resource'] = []
+            for line in buckets_list:
+                line = line.strip()
+                if line:
+                    line = line.strip()
+                    new_policy['Resource'].append(ext_policy['Resource'](line))
+            statements.append(new_policy)
+    return statements
+
+
+def _save_policy_json(policy_json, file_path):
+    with open(file_path + '.json', 'w') as file_handler:
```

Comment:
The policy.json is made once from the UploadCredentials._get_policy, EXTERNAL_BUCKET_STATEMENTS, and external_aws_s3_transfer_buckets file.  upload_credentials will look for the json first and then try to create it.  This functionality is triggered on the retry arg in  UploadCredentials._get_external_bucket_policy

deleting the policy.json will trigger a rebuild.  This allows us to update the list on a live machine.

See the test data for example buckets files.  src/encoded/tests/data/helpers/external_bucket_list_many